src/Semaphore.php

```diff
@@ -7,11 +7,10 @@
 /**
  * A non-blocking counting semaphore.
  *
- * Objects that implement this interface should guarantee that all operations
- * are atomic. Implementations do not have to guarantee that acquiring a lock
- * is first-come, first serve.
+ * Objects that implement this interface should guarantee that all operations are atomic. Implementations do not have to
+ * guarantee that acquiring a lock is first-come, first serve.
  */
-interface Semaphore
+interface Semaphore extends Mutex
```

Comment:
The separation here exists because a mutex indicates the lock should be exclusive, otherwise there would have been a single interface.

Reply:
Yes, noticed that I did it the wrong way. Mutex should extend semaphore.